models/action.go

```diff
@@ -633,6 +634,42 @@ func CommitRepoAction(opts CommitRepoActionOptions) error {
 			})
 		}
 
+		// if there is a pull request fro this branch, then update pull request
+		prs, err := GetUnmergedPullRequestsByHeadInfo(repo.ID, refName)
+		if err != nil {
+			return fmt.Errorf("GetUnmergedPullRequestsByHeadInfo: %v", err)
+		}
+
+		if err := PullRequestList(prs).LoadAttributes(); err != nil {
+			return fmt.Errorf("PullRequestList.LoadAttributes: %v", err)
+		}
+
+		var issues = make([]*Issue, 0, len(prs))
+		var lastCommitID = opts.Commits.Commits[len(opts.Commits.Commits)-1].Sha1
+		for _, pr := range prs {
+			pr.LastCommitID = lastCommitID
+			if err := pr.UpdateCols("last_commit_id"); err != nil {
+				return fmt.Errorf("UpdateCols: %v", err)
+			}
+
+			pr.Issue.PullRequest = pr
+			issues = append(issues, pr.Issue)
+		}
+
+		if _, err := IssueList(issues).LoadRepositories(); err != nil {
+			return fmt.Errorf("IssueList.LoadRepositories: %v", err)
+		}
+
+		for _, issue := range issues {
```

Comment:
Do we really need to store link to commit? I think we should generate it on the fly as it is internal url and will become invalid if user changes Gitea configuration

Reply:
If we didn't save link before, then we have to read repo table to know the repo path for every comment. I don't think that's necessary.

Reply:
The link is a relative link not an absolute link.

Reply:
yes but it will include AppSubURL that can be changed in settings. Maybe save only issue.Repo.FullName() in link and create new method for comment type that depending on comment type would generate URL (in this case AppSubURL + "/" + comment.Link + "/commit/" + comment.Sha1).
Additional change needed is to update link column on user/org rename and repository rename

Reply:
done.